aws-ssm-maintenancewindow/src/main/java/software/amazon/ssm/maintenancewindow/translator/ExceptionTranslator.java

```diff
@@ -0,0 +1,62 @@
+package software.amazon.ssm.maintenancewindow.translator;
+
+import software.amazon.awssdk.services.ssm.model.AlreadyExistsException;
+import software.amazon.awssdk.services.ssm.model.DoesNotExistException;
+import software.amazon.awssdk.services.ssm.model.ResourceLimitExceededException;
+import software.amazon.awssdk.services.ssm.model.FeatureNotAvailableException;
+import software.amazon.awssdk.services.ssm.model.TargetInUseException;
+import software.amazon.awssdk.services.ssm.model.IdempotentParameterMismatchException;
+import software.amazon.awssdk.services.ssm.model.TooManyUpdatesException;
+import software.amazon.awssdk.services.ssm.model.InternalServerErrorException;
+import software.amazon.awssdk.services.ssm.model.SsmRequest;
+import software.amazon.cloudformation.exceptions.BaseHandlerException;
+import software.amazon.cloudformation.exceptions.CfnAlreadyExistsException;
+import software.amazon.cloudformation.exceptions.CfnServiceLimitExceededException;
+import software.amazon.cloudformation.exceptions.CfnServiceInternalErrorException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnThrottlingException;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+
+/**
+ * Translator between service model exceptions and CloudFormation exceptions.
+ */
+public class ExceptionTranslator {
+
+    /**
+     * Translates service model exceptions to CloudFormation exceptions.
+     *
+     * @param serviceException Service model exception to translate.
+     * @param request Type of SsmRequest where the service exception came from.
+     * used for better exception messaging for the customer.
+     * @return CloudFormation-type exception converted from a service model exception.
+     */
+    public BaseHandlerException translateFromServiceException(final Exception serviceException,
```

Comment:
Just a question. Would it be better if we move the ExceptionTranslator to common as well? It seems like there will be some repetitions in exception among the 3 MW resources. Or maybe we could do a refactor later as well?

Reply:
Good refactoring idea. Feel free to submit a separate PR for this.